include/abb_librws/v1_0/rws_client.h

```diff
@@ -155,6 +156,18 @@ class RWSClient
                                        const std::string& tool = "",
                                        const std::string& wobj = "");
 
+
+  /**
+   * \brief Checks progress of provided resource.
+   * 
+   * \param id Resource id to check
+   *
+   * \return Returns progress of resource.
+   *
+   * \throw \a RWSError if something goes wrong.
+   */
+  progress::ProgressInfo getProgress(std::string const& id);
```

Comment:
Should not be a member function, as it does not require access to private members of `RWSClient`.